### ISSUE TITLE
Protocol Detection

### ISSUE THREAD

**Issue by amdbuilder** [modxcms/pThumb#1]
Hello,

It looks like pThumb is unable to handle resources being called sans protocol (http://www.paulirish.com/2010/the-protocol-relative-url/) which is a valid method of serving files.

In these cases it should try connecting via https and then http in my opinion, avoiding a nonsecure warning if forcing http on a page being accessed securely.

The error if it helps:

```
[2013-08-22 23:30:48] (ERROR @ /index.php) [pThumb] Resource: 170 || Image: (none)
Retrieving //domain.com/images/file.jpg
Target filename: /home/domain/public_html/assets/components/phpthumbof/cache/domain.com/images/file.jpg
cURL error: <url> malformed  *** Skipping ***
```

**Comment by oo12** [comment]
Yeah, I like protocol-relative URLs too. My ["remote images" regex](https://github.com/oo12/phpThumbOf/blob/pThumb/core/components/phpthumbof/model/phpthumbof.class.php#L95) detects them, but looks like cURL doesn't like them.

Would you try this fix?  In core/components/phpthumbof/model/phpthumbof.class.php right after [line 103](https://github.com/oo12/phpThumbOf/blob/pThumb/core/components/phpthumbof/model/phpthumbof.class.php#L103) add this:

```
if ($src[0] === '/') {
    $src = (empty($_SERVER['HTTPS']) ? 'http:' : 'https:') . $src;
}
```

Does that fix it?

One more question. Is that image on the same site as your MODX install, or is it coming from another site?  Right now pThumb makes the assumption that anything with a URL (http(s) or //) is an off-site image and uses cURL to first download and cache it, but I could make it check more thoroughly to make sure it's not in fact coming from the same site.  I don't have an idea how frequently people do this...

**Comment by amdbuilder** [comment]
That seems to have resolved the problem, thanks for the quick fix!

The images are hosted on the same server, but served using a subdomain & separate media source.  Generally speaking I think most assets are being served off the same server, with subdomains being next most popular and offsite last.

**Comment by oo12** [comment]
Glad that fixed it; I'll incorporate it in the next release.

Thanks for pointing it out.  I hadn't realized cURL couldn't handle protocol-relative URLs.

**Comment by oo12** [comment]
Fixed in 2.0.0-rc3.
